### DISCUSSION TITLE
Running ArgoCD outside of cluster

### DISCUSSION THREAD

**Discussion by malcolm061990** [argoproj/argo-cd#5571] [General]
Hi. I am just starting investigation of ArgoCD and found this project very cool :) So thanks for that.
But I have one misunderstanding. As I understand ArgoCD must run only using some k8s cluster entities like namespace,  configmaps, etc.
I want to run ArgoCD outside of k8s cluster without connecting to it for start. I will configure connection with k8s cluster later.  In [this](https://argo-cd.readthedocs.io/en/stable/developer-guide/running-locally/) article ArgoCD is launched locally but **with** connection to some cluster.
How can I run ArgoCD (in docker, for example) without connection to k8s cluster?

**Comment by wtam2018** [comment]
Argo CD is designed to be running in a K8 cluster.   You can run Argo CD servers locally or in docker containers (https://argoproj.github.io/argo-cd/developer-guide/running-locally/) but it is a setup for development only.

**Reply by malcolm061990** [reply]
Thanks. So, the best practice is to run separate ArgoCD for each cluster, isn't it? 
For example: I have five k8s clusters and 5 instances of argocd. Am I right?

**Reply by hjorth** [reply]
I run 1 ArgoCD against 8 clusters.

From the different readings available, ArgoCD can scale the Application controller and run hundreds of clusters and thousands of applications within the same instance.

So its build to scale - i havent seen anywhere any recommendation about running multiple instances of ArgoCD.

**Reply by malcolm061990** [reply]
> I run 1 ArgoCD against 8 clusters.
> 
> From the different readings available, ArgoCD can scale the Application controller and run hundreds of clusters and thousands of applications within the same instance.
> 
> So its build to scale - i havent seen anywhere any recommendation about running multiple instances of ArgoCD.

Ok. Imagine, you have three envs: test, dev, prod. Each env has 2 clusters. So you have 6 clusters.
Where is it better to deploy argocd? It's bad to deploy argocd on test or dev because in that case from NON prod clusters you have to access the prod clusters.

**Reply by hjorth** [reply]
I have an "admin" cluster.

Might not be best pratice. But we run log and monitoring federation there as well.

**Reply by malcolm061990** [reply]
> I have an "admin" cluster.
> 
> Might not be best pratice. But we run log and monitoring federation there as well.

Thanks, yes it can be not a bad point...
But it's interesting to hear the best practice from the developer's side.

**Reply by rumstead** [reply]
> I have an "admin" cluster.
> 
> Might not be best pratice. But we run log and monitoring federation there as well.

We do this as well minus the log and monitoring.

**Comment by hjorth** [comment]
My first test with ArgoCD, was locally on microk8s.

Worked perfectly fine.

Maybe try that approach.
